chapters/en/slides/module2/module2_03.Rmd

```diff
@@ -0,0 +1,284 @@
+---
+params:
+  dynamictitle: "module2_03"
+title: "`r params$dynamictitle`"
+output:
+  md_document:
+    variant: gfm
+    pandoc_args: "--markdown-headings=atx"
+---
+
+```{r setup, include=FALSE}
+source('../../../../slide-setup.R')
+# Remember to also manually update the YAML title above
+```
+
+type: slides
+
+# Creating bar charts and histograms
+
+Notes:
+In this module
+we will be learning how to create bars charts for counting observations,
+and a special types of bar chars called histograms
+for showing distributions of data.
+
+---
+
+## Subsetting the global development data
+
+```{python}
+import pandas as pd
+
+
+url = 'https://raw.githubusercontent.com/UofTCoders/workshops-dc-py/master/data/processed/world-data-gapminder.csv'
+gm = pd.read_csv(url, parse_dates=['year'])
+gm2018 = gm[gm['year'] == '2018']
+gm2018
+```
+
+Notes:
+In this slide deck,
+we will continue to work with the gapminder dataset.
+To make it more suitable for this slide deck
+let's subset it to include observations from the year 2018 only.
+
+When filtering on a date in pandas,
+we can either type out just the year portion
+or the entire date string in the format shown in the slide.
+
+---
+
+## Creating a bar chart of the total population per continent
+
+```{python}
+import altair as alt
+
+
+alt.Chart(gm2018).mark_bar().encode(
+    x='region',
+    y='sum(population)')
+```
+
+Notes:
+Bar charts are often avoided when showing summary statistics such as the mean
+because they hide any variation,
+something we will talk more about in later modules.
+
+But they are fine for showing statistics
+such as counts of observations, summations, or individual values
+(e.g. the number of people in a country for a specific year).
+
+Here we create a bar chart of a single value per country,
+which represents the sum of all the countries' populations.
+
+---
+
+## Horizontal bar charts are made by switching the x and y encoding
+
+```{python}
+alt.Chart(gm2018).mark_bar().encode(
+    y='region',
+    x='sum(population)')
+```
+
+Notes:
+If we switched x and y,
+we would create a horizontal bar chart instead.
+
+Although vertical bar charts are more commonly seen,
+horizontal bar charts are preferred
+when the labels on the categorical axis
+become so long that they need to be rotated
+to be readable in a vertical bar chart.
+
+Since this is the case for our plot
+we will continue to use horizontal bar charts
+in this slide deck.
+
+---
+
+## Visualizing counts of values with bar charts
+
+```{python}
+alt.Chart(gm2018).mark_bar().encode(
+    x='count()',
+    y='region')
+```
+
+Notes:
```

Comment:
Can we start with a scenario/question? It is not clear what a record is and why we would want to visualize these (I am guessing these are actually countries? And so count of records is actually number of countries on each continent?